MovingBlocks/Terasology
# NUI editor icons

### Contains

An implementation for rzats/gsoc-nui-editor#7.

- The NUI editor is modified to use the `StringIconTextRenderer` conveniently created in #2270, which is slightly tweaked to support `null` icon textures.
- 16x16 icons for all core widgets (`UI*`), layouts and JSON node types are added.
- An option is added to disable icons, wrapped within a new `NUIEditorConfig` class. 
- `TreeViewTestScreen` is removed as obsolete.

![editor-icons](https://cloud.githubusercontent.com/assets/13783592/16897121/9ae1dbb4-4bb0-11e6-94cd-e602b54f28b9.png)

### How to test

Open the NUI editor (F10), check out icons for widget nodes, then try enabling/disabling icons within the Settings menu.

### Outstanding before merging

- [x] Provide attribution for the icon set used ([Fugue Icons](http://p.yusukekamiyamane.com/)), as well as any other sets. @Cervator: should the attribution notices be located in `CREDITS.md`?
- [x] Add icons for common properties (`type`, `layoutInfo`, `contents`, `@LayoutConfig`-annotated properties of core widgets).
- [x] If possible, add icons for core layouts.
